### ISSUE TITLE
Segmentation fault when using files that have '.' in the name (win32)

### ISSUE THREAD

**Issue by jmathew** [jqlang/jq#1575]
I'm using the win32 on git-bash (MSYS) on Windows 10 x64.

```
jmathew@host MINGW64 ~/work/someproject (someBranch)
$ ./scripts/jq-win32.exe '.' folder1/folder2/local.settings.json
Segmentation fault


jmathew@host MINGW64 ~/work/someproject (someBranch)
$ mv  folder1/folder2/local.settings.json  folder1/folder2/local.json

jmathew@host MINGW64 ~/work/someproject (someBranch)
$ ./scripts/jq-win32.exe  '.' folder1/folder2/local.json
{
    "someKey":"some value"
}
```

JQ Version
```
$ ./scripts/jq-win32.exe
jq - commandline JSON processor [version 1.5]
...
```

**Comment by nicowilliams** [comment]
Thanks for the report!

**Comment by sfc-tbrown** [comment]
It looks like it has more to do with length than with content:

```
tbrown@local ~/dev/jq/1575
$ jq '.' folder1/folder2/local.json
{
  "someKey": "some value"
}

tbrown@local ~/dev/jq/1575
$ mv folder1/folder2/local.json folder1/folder2/loc.al.json

tbrown@local ~/dev/jq/1575
$ jq '.' folder1/folder2/loc.al.json
{
  "someKey": "some value"
}

tbrown@local ~/dev/jq/1575
$ mv folder1/folder2/loc.al.json folder1/folder2/l.oc.al.json

tbrown@local ~/dev/jq/1575
$ jq '.' folder1/folder2/l.oc.al.json
Segmentation fault

tbrown@local ~/dev/jq/1575
$ mv folder1/folder2/l.oc.al.json folder1/folder2/loocaal.json

tbrown@local ~/dev/jq/1575
$ jq '.' folder1/folder2/loocaal.json
Segmentation fault
```

Version:
```
$ jq --version
jq-1.5
```

I have the 64-bit version IIRC

**Comment by pkoppstein** [comment]
The long-pathname problem was fixed on Sept 11, 2015 (63dd033), shortly AFTER
the release of jq 1.5 :-(

You can get the latest jq.exe file by following the instructions at https://github.com/stedolan/jq/wiki/Installation#windows-using-appveyor
